webui/src/lib/api/index.js

```diff
@@ -670,12 +670,15 @@ export const uploadWithProgress = (url, file, method = 'POST', onProgress = null
             resolve({
                 status: xhr.status,
                 body: xhr.responseText,
-                contentType: xhr.getResponseHeader('Content-Type'),
-                etag: xhr.getResponseHeader('ETag'),
-                contentMD5: xhr.getResponseHeader('Content-MD5'),
-            })
+                rawHeaders: xhr.getAllResponseHeaders(), // add raw headers 
+            });
         });
-        xhr.addEventListener('error', () => reject(new Error('Upload Failed')));
+        xhr.addEventListener('error', () => resolve({
```

Comment:
The reason for this is that I wanted to throw the correct error. This way if there is an error due to missing CORS setting the calling func will catch it and throw suitable error. In case other piece of code uses this func and upload fails for some reason - the error will just state 'upload failed'.

Reply:
The proper way of handling errors in Promises is using the try/catch mechanism at the calling function.
This should still reject, and the error (the object returned to the reject function) can be analyzed further by the calling function.
I would like to mention in this context that this is javascript and it's s**t.